File: Algorithms/grokking algorithms/4. Quick Sort/euclidean.js
const euclidean = (a, b) => {
    if (a === 0 || b === 0) {
        return 0
    }
    const remainder = a%b
    if (remainder === 0){
        return b
    }
    return euclidean(b, remainder)
}
console.log(euclidean(270, 192))

File: Algorithms/grokking algorithms/4. Quick Sort/euclidean.go
package main

import "fmt"

func euclidean(a int, b int) int {
	if a == 0 || b == 0 {
		return 0
	}
	remainder := a % b
	if remainder == 0 {
		return b
	}
	return euclidean(b, remainder)
}

func main() {
	gcd := euclidean(270, 192)
	fmt.Println(gcd)
}
